xla/service/gpu/triton_autotuner.cc

```diff
@@ -504,11 +521,61 @@ absl::StatusOr<std::unique_ptr<HloModule>> CublasGemmAutotuneExtractor(
   return new_module;
 }
 
+absl::StatusOr<std::unique_ptr<HloModule>> CudnnGemmAutotuneExtractor(
+    const AutotuneConfig& autotune_config, const HloFusionInstruction* fusion,
+    const DebugOptions& debug_opts, const int plan_id) {
+  std::unique_ptr<HloModule> new_module =
+      ExtractInstructionIntoNewModule(*fusion);
+  new_module->mutable_config().set_debug_options(debug_opts);
+
+  GpuBackendConfig gpu_config;
+  FusionBackendConfig& backend_config =
+      *gpu_config.mutable_fusion_backend_config();
+  backend_config.set_kind(std::string(kCuDnnFusionKind));
+  // Provided a plan ID the autotuner just compiles one plan.
+  backend_config.mutable_cudnn_fusion_config()->set_plan_id(plan_id);
+  TF_RETURN_IF_ERROR(
+      new_module->entry_computation()->root_instruction()->set_backend_config(
+          gpu_config));
+  CuDnnFusionCompiler compiler(autotune_config);
+  TF_RETURN_IF_ERROR(compiler.Run(new_module.get()).status());
+
+  return new_module;
+}
+
 bool ShouldAllowFilteringKernelsSpillingRegisters(
     const GemmConfigSet& gemm_config_set) {
   return gemm_config_set.configs.size() > 1;
 }
 
+bool IsFusionKind(const HloInstruction& hlo, absl::string_view kind) {
+  auto gpu_config = hlo.backend_config<GpuBackendConfig>();
+  if (!gpu_config.ok()) {
+    return false;
+  }
+  return gpu_config->fusion_backend_config().kind() == kind;
+}
+
+std::vector<int64_t> GetCuDnnPlanIds(const HloInstruction& hlo,
+                                     const AutotuneConfig& autotune_config) {
+  auto gpu_config = hlo.backend_config<GpuBackendConfig>();
+  if (!gpu_config.ok() ||
+      gpu_config->fusion_backend_config().has_cudnn_fusion_config()) {
+    return {};
+  }
+  std::vector<int64_t> result(
+      CuDnnFusionCompiler(autotune_config)
+          .GetAvailablePlanCount(*DynCast<HloFusionInstruction>(&hlo)));
+  std::iota(result.begin(), result.end(), 0);
```

Comment:
Do you need this `iota`? When you take the size of the vector below, you could just use the plan count directly. When you iterate over the plans, a for loop should do.

Reply:
Changed.